compact_str/src/repr/discriminant.rs

```diff
@@ -21,7 +19,15 @@ pub struct DiscriminantMask {
 impl DiscriminantMask {
     #[inline(always)]
     pub const fn discriminant(&self) -> Discriminant {
-        if self.val == HEAP_MASK {
+        if self.val == u8::MAX {
+            // HACK: This allows us to make assertions within a `const fn` without requiring
+            // nightly, see unstable `const_panic` feature. This results in a build
+            // failure, not a runtime panic
```

Comment:
`const_panic` was stabilized in 1.57; it's probably worth it to require 1.57 since we're bumping the MSRV anyway.

Also, to be clear: this can still be a runtime panic if the function is called at runtime (not in a const context).

Reply:
This makes sense to me! Updated the MSRV to 1.57 and now just call `panic!(...)`